### PR TITLE
Allow named map definitions (issue #231)

### PR DESCRIPTION
**Pull request by SeanHeelan**
Prior to this, map definitions must be in sections named "maps". This
commit adds support for named map definitions matching
"maps/<map-name>".

More context in #231 

**Before** 

```
[sean@localhost ebpf-verifier]$ ./check /home/sean/go/src/github.com/optimyze/prodfiler/pf-host-agent/support/ebpf/tracer.ebpf tracepoint/syscalls/sys_enter_bpf --domain=zoneCrab
error: Can't find any maps sections in file /home/sean/go/src/github.com/optimyze/prodfiler/pf-host-agent/support/ebpf/tracer.ebpf
```

**After**

```
[sean@localhost ebpf-verifier]$ ./check /home/sean/go/src/github.com/optimyze/prodfiler/pf-host-agent/support/ebpf/tracer.ebpf tracepoint/syscalls/sys_enter_bpf --domain=zoneCrab
1,0.046412,4928
```

@elazarg I've marked this as 'draft' as I wasn't entirely sure how to add a test in the current repo setup. The approach appears to be to add a _compiled_ ELF file containing the required test sections, and then update [test_verify.cpp](https://github.com/vbpf/ebpf-verifier/blob/master/src/test/test_verify.cpp). Is that correct, or should I do something else? 

(Also feel free to leave any comments you may have on the existing code. I haven't wrote any C++ in about 6 years and the verifier codebase is still largely an unknown to me, so apologies for any potential insanity ;))

### CODE REVIEW

File: src/asm_files.cpp [hunk 1]
```diff
@@ -47,6 +48,27 @@ EbpfMapDescriptor* find_map_descriptor(int map_fd) {
...
+    std::string maps_prefix = "maps/";
+    for (ELFIO::Elf_Half i = 0; i < reader.sections.size(); ++i) {
+        s = reader.sections[i];
+        auto name = s->get_name();
+        if (name.length() > 5 && name.compare(0, maps_prefix.length(), maps_prefix) == 0) {
+            platform->parse_maps_section(map_descriptors, s->get_data(), s->get_size(), platform, *options);
+            section_indices.insert(s->get_index());
+        }
+    }
+}
```
Comment (elazarg) [inline src/asm_files.cpp:70]: I think it's better to merge both branches, so a valid name has either `'\0'` or `'/'` in the 5th index.

Also, it should be then possible to encapsulate the decision in a dedicated function `is_map_section`, accepting a string and returning a boolean.

**Comment by elazarg** [comment]
@SeanHeelan is this ready?

File: src/asm_files.cpp (hunk 1 above)
Comment (SeanHeelan) [reply src/asm_files.cpp:67]: Done

**Comment by SeanHeelan** [comment]
> @SeanHeelan is this ready?

@elazarg Apologies for the radio-silence on this. Got distracted by work ;) So the PR is ready in terms of the functionality I wanted to add is there and works, but my original question on testing still stands. It was:

```
@elazarg I've marked this as 'draft' as I wasn't entirely sure how to add a test in the current repo setup. The approach appears to be to add a compiled ELF file containing the required test sections, and then update test_verify.cpp. Is that correct, or should I do something else?
```

I ask as I'm reluctant to push ELF files that do not come with the original source file + a mechanism for building the ELF. However, if that is the approach currently taken I will go along with it of course =)

**Comment by elazarg** [comment]
@SeanHeelan this is a good point. Currently we add tests by first adding a C file _and_ a compiled ELF file to [vbpf/ebpf-samples](https://github.com/vbpf/ebpf-samples) submodule, and in the PR we bump the version of the submodule and add a test using that file. See PR https://github.com/vbpf/ebpf-samples/pull/10 for example.
The reason for attaching the compiled version is to avoid differences between compilers to affect the tests.

This workflow is somewhat awkward but for this specific PR it's also the best IMO.

**Comment by dthaler** [comment]
Also add to ebpf-samples/CMakeLists.txt the ability to build the .o file.  That step is manual, not invoked from the root CMakeLists.txt

**Comment by elazarg** [comment]
@SeanHeelan thanks for the PR!

**Pull request merged**